[BUG] Query not working anymore with Memgraph 2.9

**Memgraph version**
2.9

**Environment**
Memgraph 2.9 running in Docker, storage mode IN_MEMORY_ANALYTICAL
Python 3.9
Neo4j driver for Python using the Bolt interface (tested with v5.7 and v5.10)

<details>
  <summary>Memgraph Configuration</summary>

```
--bolt-server-name-for-init=Neo4j/
--log-file=/var/log/memgraph/memgraph.log
--log-level=WARNING
--memory-limit=0
--metrics-address=0.0.0.0
--metrics-port=2004
--query-modules-directory=/usr/lib/memgraph/query_modules
--storage-parallel-index-recovery=false
--storage-properties-on-edges=true
--storage-recover-on-startup=true
--storage-snapshot-interval-sec=300
--storage-snapshot-on-exit=true
--storage-snapshot-retention-count=3
--storage-wal-enabled=true
--data-directory=/var/lib/memgraph
--isolation-level=READ_COMMITTED
--telemetry-enabled=true
```

</details>

**Describe the bug**
When we execute the following query in Memgraph 2.9, the python driver is complaining that the data returned is not valid

```cypher
MATCH p = (s)-[:IS_RELATED*..2]-(d)
WHERE ID(s) = 11028 AND ID(d) = 11040
RETURN p
```

```python 
self = <Record p=<neo4j._codec.hydration._common.BrokenHydrationObject object at 0x10b3c0fa0>>, index = 0

    def _super_getitem_single(self, index):
        value = super().__getitem__(index)
        if isinstance(value, BrokenHydrationObject):
>           raise self._broken_record_error(index) from value.error
E           neo4j.exceptions.BrokenRecordError: Record contains broken data at 0 ('p')
```

**To Reproduce**
Steps to reproduce the behavior:
1. Run the following query
```cypher
MATCH p = (s)-[:IS_RELATED*..2]-(d)
WHERE ID(s) = 11028 AND ID(d) = 11040
RETURN p
```


**Expected behavior**
The exact same code/query is working fine with Memgraph 2.8
When I run the same query on Memgraph-lab, it seems to work as well

**Logs**

**Additional context**
Here is the result when i execute the same query from Memgraph-lab 
```json
{
    "p": {
        "nodes": [
            {
                "id": 11028,
                "labels": [
                    "Node",
                    "BuiltinTag"
                ],
                "properties": {
                    "kind": "BuiltinTag",
                    "uuid": "fb2406d6-dd44-4216-ae09-f8e89377e2e2"
                },
                "type": "node"
            },
            {
                "id": 11045,
                "labels": [
                    "Relationship"
                ],
                "properties": {
                    "name": "builtintag__testperson",
                    "uuid": "9a551ed0-933c-4e01-8fb2-179dcc87e81a"
                },
                "type": "node"
            },
            {
                "id": 11040,
                "labels": [
                    "Node",
                    "TestPerson"
                ],
                "properties": {
                    "kind": "TestPerson",
                    "uuid": "705badb3-661f-4b70-9f14-2bcd9ec6aa45"
                },
                "type": "node"
            }
        ],
        "relationships": [
            {
                "id": 36897,
                "start": 11028,
                "end": 11045,
                "label": "IS_RELATED",
                "properties": {
                    "branch": "main",
                    "branch_level": 1,
                    "from": "2023-07-25T05:16:44.370399Z",
                    "status": "active"
                },
                "type": "relationship"
            },
            {
                "id": 36896,
                "start": 11040,
                "end": 11045,
                "label": "IS_RELATED",
                "properties": {
                    "branch": "main",
                    "branch_level": 1,
                    "from": "2023-07-25T05:16:44.370399Z",
                    "status": "active"
                },
                "type": "relationship"
            }
        ],
        "type": "path"
    }
}
```

--- a/src/communication/bolt/v1/value.hpp
+++ b/src/communication/bolt/v1/value.hpp
@@ -112,7 +112,7 @@ struct Path {
     for (uint i = 0; i < edges.size(); i++) {
       const auto &e = edges[i];
       const auto &v = vertices[i + 1];
-      UnboundedEdge unbounded_edge{e.id, e.type, e.properties};
+      UnboundedEdge unbounded_edge{e.id, e.type, e.properties, e.element_id};
       add_element(this->edges, unbounded_edge, e.to == v.id ? 1 : -1, 1);
       add_element(this->vertices, v, 1, 0);
     }

--- a/tests/drivers/python/v5_8/path.py
+++ b/tests/drivers/python/v5_8/path.py
@@ -1,0 +1,36 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Memgraph Ltd.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+# License, and you may not use this file except in compliance with the Business Source License.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0, included in the file
+# licenses/APL.txt.
+
+from neo4j import GraphDatabase, basic_auth
+from neo4j.exceptions import ClientError, TransientError
+
+with GraphDatabase.driver("bolt://localhost:7687", auth=None, encrypted=False) as driver:
+    # Setup the graph.
+    with driver.session() as session:
+        session.run("MATCH (n) DETACH DELETE n").consume()
+        session.run(
+            "CREATE (:Node{id:1})-[:IS_CONNECTED]->(:Node{id:2})-[:IS_CONNECTED]->(:Node{id:3})-[:IS_CONNECTED]->(:Node{id:4})"
+        ).consume()
+
+    # Return a path of length 2
+    with driver.session() as session:
+        res = session.run("MATCH p = (:Node)-[:IS_CONNECTED*..2]->(:Node) RETURN p").values()
+        assert len(res) == 5, "Didn't find all the paths"
+
+    # Return a path of length 3
+    with driver.session() as session:
+        res = session.run("MATCH p = (:Node)-[:IS_CONNECTED*3]->(:Node) RETURN p").values()
+        assert len(res) == 1, "Didn't find the path"
+
+    print("All ok!")

--- a/tests/drivers/python/v5_8/run.sh
+++ b/tests/drivers/python/v5_8/run.sh
@@ -24,3 +24,4 @@ source ve3/bin/activate
 python3 docs_how_to_query.py || exit 1
 python3 max_query_length.py || exit 1
 python3 transactions.py || exit 1
+python3 path.py || exit 1

